app/controllers/community/appointments_controller.rb

```diff
@@ -16,66 +18,62 @@ def home
     end
 
     # Reschedules appointment (only if patient already has one scheduled)
-    # rubocop:disable Metrics/AbcSize
     def index
-      @appointment = current_patient.appointments.not_checked_in.current
-      if @appointment && !@appointment.can_cancel_and_reschedule?
-        return redirect_to(
-          home_community_appointments_path,
-          flash: {
-            alert: t('alerts.cannot_cancel_or_reschedule', days: Rails.configuration.x.schedule_up_to_days),
-            cy: 'cannotCancelOrRescheduleText'
-          }
-        )
-      end
+      appointment_can_cancel_and_reschedule
+
+      # If patient already had a dose, keep it in the same UBS.
+      # This is an optimized query, hence a little odd using +pick+s.
+      ubs_id = Appointment.where(id: current_patient.doses.pick(:appointment_id)).pick(:ubs_id)
 
       @days = parse_days
       @appointments = scheduler.open_times_per_ubs(from: @days.days.from_now.beginning_of_day,
-                                                   to: @days.days.from_now.end_of_day)
+                                                   to: @days.days.from_now.end_of_day,
+                                                   filter_ubs_id: ubs_id)
     rescue AppointmentScheduler::NoFreeSlotsAhead
       redirect_to home_community_appointments_path, flash: { alert: 'Não há vagas disponíveis para reagendamento.' }
     end
 
-    # rubocop:enable Metrics/AbcSize
-
     def create
+      appointment_can_cancel_and_reschedule
+
+      # If patient already had a dose, keep it in the same UBS
+      ubs_id = current_patient.doses.first&.appointment&.ubs_id
```

Comment:
O que tu acha da gente deixar estourar no doses mesmo?

Caso contrário o erro vai ser assim na hora de criar o agendamento de 2ª dose: 
```
ActiveRecord::RecordInvalid: Validation failed: Ubs é obrigatório(a)
```

Talvez confunda quem for investigar

Reply:
Desculpa, não entendi. Estourar o que?

Reply:
Se chegar `ubs_id = nil` por alguma razão desconhecida, esse dado vai trafegar até o scheduler e quebrar lá. Acho que a gente como tem contexto vai pensar na possível ausência de doses, mas podemos esquecer tb haha. Aí talvez deixar estourar aqui mesmo. Por exemplo: 

(n lembro como é o erro :sweat_smile:)

`function first called on nil` ~ `current_patient.[].first`